File: synthesis/src/schemas/connector.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, ConfigDict, Field


class ConnectorRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    organization_id: UUID
    type: str
    name: str
    enabled: bool
    auto_synthesize: bool
    config: dict = Field(default_factory=dict)
    credentials: dict = Field(default_factory=dict)
    last_sync_at: datetime | None = None
    last_sync_error: str | None = None
    created_at: datetime
    updated_at: datetime


class ConnectorCreate(BaseModel):
    type: str
    name: str
    enabled: bool = True
    auto_synthesize: bool = True
    config: dict = Field(default_factory=dict)
    credentials: dict = Field(default_factory=dict)


class ConnectorUpdate(BaseModel):
    name: str | None = None
    enabled: bool | None = None
    auto_synthesize: bool | None = None
    config: dict | None = None
    credentials: dict | None = None

File: connector/src/types/connector.ts
export interface Connector {
  id: string;
  organization_id: string;
  type: string;
  name: string;
  enabled: boolean;
  auto_synthesize: boolean;
  config: Record<string, unknown>;
  credentials: Record<string, unknown>;
  last_sync_at?: string | null;
  last_sync_error?: string | null;
  created_at: string;
  updated_at: string;
}

export interface ConnectorConfigField {
  key: string;
  label: string;
  type: "text" | "number" | "boolean" | "multi_text" | "multi_select" | "select";
  required?: boolean;
  default?: unknown;
  options?: string[];
  help?: string;
}

export interface ConnectorCatalogItem {
  type: string;
  display_name: string;
  description: string;
  auth_method: "oauth2" | "api_key" | string;
  category: "input" | "output";
  icon: string;
  available?: boolean;
  missing_env_vars?: string[];
  config_fields: ConnectorConfigField[];
}

export interface ConnectorCreatePayload {
  type: string;
  name: string;
  enabled?: boolean;
  auto_synthesize?: boolean;
  config?: Record<string, unknown>;
  credentials?: Record<string, unknown>;
}
